### PR TITLE
[k8s] Add optional annotations and labels for CatalogSource object

### PR DESCRIPTION
**Pull request by m-amaresh**
Summary: This PR will add optional annotations and labels to CatalogSource object for pixie-operator-chart, so Cluster Admins can add annotation and labels to resources created by CatalogSource `pixie-operator-index`

Relevant Issues: https://github.com/pixie-io/pixie/issues/1118

Type of change: /kind cleanup

Test Plan: None

### CODE REVIEW

**Comment by JamesMBartlett** [comment]
@pixie-io-buildbot test this please

**Comment by JamesMBartlett** [comment]
@pixie-io-buildbot test this please

File: k8s/operator/helm/templates/02_catalog.yaml [hunk 1]
```diff
@@ -3,6 +3,12 @@ kind: CatalogSource
 metadata:
   name: pixie-operator-index
   namespace: {{ .Values.olmOperatorNamespace }}
+  {{- if .Values.pod.annotations }}
```
[the remark this answers is not in the document]
Comment (m-amaresh) [reply k8s/operator/helm/templates/02_catalog.yaml]: Hi, Thanks for your review. It was a silly mistake, I have fixed.

File: k8s/operator/helm/values.yaml [hunk 2]
```diff
@@ -15,7 +15,12 @@ olmOperatorNamespace: "px-operator"
 # The bundle channel which OLM should listen to for the Vizier operator bundles.
 # Should be "stable" for production-versions of the operator, and "test" for release candidates.
 olmBundleChannel: "stable"
-
+# Optional annotations and labels for CatalogSource.
+catalogsource:
```
[the remark this answers is not in the document]
Comment (m-amaresh) [reply k8s/operator/helm/values.yaml]: Yes, good idea, Changed it

File: k8s/operator/helm/templates/02_catalog.yaml [hunk 3]
```diff
@@ -3,6 +3,12 @@ kind: CatalogSource
...
+  annotations: {{ .Values.catalogsource.annotations | toYaml | nindent 4 }}
+  {{- end }}
+  {{- if .Values.pod.annotations }}
```
[the remark this answers is not in the document]
Comment (m-amaresh) [reply k8s/operator/helm/templates/02_catalog.yaml]: Fixed

**Comment by m-amaresh** [comment]
hello @aimichelle , Thanks for reviewing the PR. I have updated the file(s)
Can you please review again.

**Comment by aimichelle** [comment]
@pixie-io-buildbot test this please

**Review by aimichelle — APPROVED** [review]

**Pull request merged**